evm-contracts/src/v0.7/dev/StalenessFlaggingValidator.sol

```diff
@@ -108,11 +91,35 @@ contract StalenessFlaggingValidator is ConfirmedOwner {
    * @return address[] memory stale aggregators
    */
   function update(address[] calldata aggregators) external returns (address[] memory){
-    address[] memory staleAggregators = check(aggregators);
+    address[] memory staleAggregators = checkStalenessOf(aggregators);
     s_flags.raiseFlags(staleAggregators);
     return staleAggregators;
   }
 
+  /**
+   * @notice Check for staleness in an array of aggregators
+   * @dev Overriding KeeperInterface
+   * @param data bytes encoded address array
+   * @return needsUpkeep bool indicating whether upkeep needs to be performed
+   * @return staleAggregators bytes encoded address array of stale aggregator addresses
+   */
+  function checkForUpkeep(bytes calldata data) external view override returns (bool, bytes memory) {
+    address[] memory staleAggregators = checkStalenessOf(abi.decode(data, (address[])));
+    bool needsUpkeep = (staleAggregators.length > 0);
+    return (needsUpkeep, abi.encode(staleAggregators));
+  }
+
+  /**
+   * @notice Check for staleness in an array of aggregators, raise a flag
+   * on the flags contract for each aggregator that is stale
+   * @dev Overriding KeeperInterface
+   * @param data bytes encoded address array
+   */
+  function performUpkeep(bytes calldata data) external override {
```

Comment:
Could you make this work by calling the `update` method? They are almost identical.